solr/core/src/java/org/apache/solr/handler/admin/CoreAdminHandler.java

```diff
@@ -546,6 +559,8 @@ public static class TaskObject {
       final Callable<SolrQueryResponse> task;
       public String rspInfo;
       public Object operationRspInfo;
+      private volatile String status;
+      private volatile boolean polled;
```

Comment:
Could use javadoc.  From observing you code, this is set to true once the status has been requested AND it is no longer RUNNING.  Maybe a different name like `polledAfterCompletion` would be clearer.